### ISSUE TITLE
Date formats

### ISSUE THREAD

**Issue by FichteFoll** [Bionus/imgbrd-grabber#1968]
**Bug description**

Grabber currently uses multiple formats for dates, neither of which respects by `LC_TIME` locale setting.

The following screenshot shows the date as seen in the favorites edit dialog on the left and in the "more details" popup of an image on the right.

![2020-05-21_16-06-40](https://user-images.githubusercontent.com/931051/82567086-2e67e700-9b7d-11ea-9480-15805ff903dc.png)

I would like to see the RFC-3339 (or ISO-8601) format, as is used in the en_DK locale:

```
/tmp λ date +%x
2020-05-21

/tmp Ξ echo $LC_TIME
en_DK.UTF-8
```

**System information**

- OS: Arch Linux
- Grabber version: 7.2.1

**Comment by Bionus** [comment]
Indeed, the date formats are a little bit inconsistent.
This is due to multiple reasons, so I'll push a few fixes for that.

In your screenshots however, you're showing the "Last view" field of what I believe if a favorite's window. Date and time editors use Qt's default format, which is similar to the ISO one ("2017/06/17 18:32:09"), and I'm not sure it's useful to change it.

All other places where dates are _displayed_, however, should use the locale's setting (such as your right's screenshot).

**Comment by FichteFoll** [comment]
Can't Qt's control be [configured](https://doc.qt.io/qt-5/qdatetimeedit.html#displayFormat-prop)?

**Comment by Bionus** [comment]
They can. What I was saying is that it might not be necessary/useful to follow an user's locale for a date editor, when the default format is ISO.

For display, it will use the user's settings as long as you choose the correct language in Grabber's options.

**Comment by FichteFoll** [comment]
Unfortunately, language does not directly correlate to the preferred date format. Choosing English, which is a natual choice for anyone able to understand that language, causes `DD/MM/YYYY HH.mm` to be used, which is workable, if you remember that, but will always have me second guessing whether it might be `MM/DD/YYYY`, because that is just as likely. I'm okay with the Qt control because I don't think `YYYY/DD/MM` is a thing.

**Comment by Bionus** [comment]
Not sure what you mean. If your system is in english and you choose the english language, it'll use your system's locale to display dates. `XX/XX/YYYY` will always be ambiguous but the only other option is to use ISO but that's not user-friendly.

**Comment by FichteFoll** [comment]
Is there no way to have Qt respect my `LC_TIME` setting?

**Comment by Bionus** [comment]
AFAIK Qt already uses LC_TIME:
https://github.com/openwebos/qt/blob/master/src/corelib/tools/qlocale_unix.cpp

Are you using the latest version from the `develop` branch?

Maybe your Qt version is too old? The `en_DK` locale you mention seem to be absent on earlier versions (apparently you'd need Qt 5.7):
https://bugreports.qt.io/browse/QTBUG-35692?focusedCommentId=327974&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel

**Comment by FichteFoll** [comment]
> Are you using the latest version from the `develop` branch?

Yes, I built fresh from the `develop` branch.

> Maybe your Qt version is too old? The `en_DK` locale you mention seem to be absent on earlier versions (apparently you'd need Qt 5.7):

"About Qt" shows 5.14.2.

**Comment by FichteFoll** [comment]
See also my locale output, if that helps:
```
~/tmp/pkg ∀ locale
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME=en_DK.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=de_DE.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT=de_DE.UTF-8
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

**Comment by Bionus** [comment]
Actually, I don't see why would the `en_DK` use the `yyyy-MM-dd` format.

In the CLDR, there are no overrides for the date format itself:
https://github.com/unicode-org/cldr/blob/master/common/main/en_DK.xml

As such, it will fallback to the `en` format:
https://github.com/unicode-org/cldr/blob/master/common/main/en.xml

See here for pattern format:
http://cldr.unicode.org/translation/date-time-1/date-time

It would seem that the `date` tool on its end uses some different `en_DK` locale, which is not the same as the one used by Qt, which is based on the CLDR.

From what I can guess, it must be based on `glibc` whose locales are terribly out-of-date:
https://github.com/lattera/glibc/blob/master/localedata/locales/en_DK

**Comment by Bionus** [comment]
Actually, you'll find more answers about this if you google the same issue for KDE, as they are also using Qt for everything, and as such also use the CLDR. People seem to suggest the `en_SE` locale, which is ISO-compliant.

However, it doesn't seem to exist in `glibc`, so I don't really know how that would work. Maybe just start Grabber with something like:
```
LC_TIME="en_SE.UTF-8" ./Grabber
```

**Comment by FichteFoll** [comment]
Oh, Qt is using CLDR? Then I understand why that is a problem because I've recently had [that issue with Thunderbird as well](https://bugzilla.mozilla.org/show_bug.cgi?id=1426907). Apparently, no locale setting for ISO date (or RFC3339, as that allows a space instead of `T` to separate date and time) [exists for that](https://unicode-org.atlassian.net/browse/CLDR-12027). I've started using `lt_LT` for Thunderbird, which works well as long as no long format (with words) is being used. I've spent an unreasonable amount of time in trying to find the best workaround and ended up with that.

I can confirm that launching with `env LC_TIME=lt_LT.UTF-8 Grabber` works. I'll patch/override my local `Grabber.desktop`, then.
